assets/js/dist/fade.js

```diff
@@ -1,40 +1,38 @@
 "use strict";
 
-$(document).ready(function () {
-  var mobileOffset = 270; // Start with only work section showing
+var mobileOffset = 270; // Start with nav-work button highlighted and only work section showing
 
-  $('#nav-work').removeClass('default-btn').addClass('current-btn');
-  $('#work').addClass('display');
-  var elements = document.getElementsByClassName('nav-btn');
-  Array.from(elements).forEach(function (el) {
-    el.addEventListener('click', function (e) {
-      e.preventDefault();
-      $('li').removeClass('current-btn');
-      $("#".concat(e.target.id)).addClass('current-btn');
-      $('#fade-wrapper').fadeOut(500, function () {
-        $('section').removeClass('display');
-        var sectionId = e.target.id.slice(4);
-        $("#".concat(sectionId)).addClass('display');
+document.getElementById('nav-work').classList.add('current-btn');
+document.getElementById('work').classList.add('display');
+var navElements = document.getElementsByClassName('nav-btn');
+Array.from(navElements).forEach(function (navElement) {
+  navElement.addEventListener('click', function (e) {
+    e.preventDefault();
+    Array.from(document.getElementsByTagName('li')).forEach(function (el) {
+      return el.classList.remove('current-btn');
+    });
+    document.getElementById(e.target.id).classList.add('current-btn');
+    $('#fade-wrapper').fadeOut(500, function () {
+      Array.from(document.getElementsByTagName('section')).forEach(function (el) {
+        return el.classList.remove('display');
+      });
+      document.getElementById(e.target.id.slice(4)).classList.add('display');
 
-        if ($(window).scrollTop() > mobileOffset) {
-          console.log('reset window');
-          window.scrollTo(0, mobileOffset);
-        }
+      if (window.scrollY > mobileOffset) {
+        window.scrollTo(0, mobileOffset);
+      }
 
-        $('#fade-wrapper').fadeIn(500, function () {
-          console.log("faded in");
-        });
-      });
+      $('#fade-wrapper').fadeIn(500);
     });
-  }); // Watch for scroll for fixed nav bar
-
-  $(window).scroll(function () {
-    if ($(window).scrollTop() > mobileOffset) {
-      $('nav').addClass('fixed-nav');
-      $('#wrapper').css('margin-top', '80px');
-    } else {
-      $('nav').removeClass('fixed-nav');
-      $('#wrapper').css('margin-top', '0');
-    }
   });
+}); // Watch for scroll for fixed nav bar
+
+window.addEventListener('scroll', function () {
+  if (window.scrollY > mobileOffset) {
+    document.getElementsByTagName('nav')[0].classList.add('fixed-nav');
+    document.getElementsByClassName('wrapper')[0].classList.add('wrapper-nav-offset');
+  } else {
+    document.getElementsByTagName('nav')[0].classList.remove('fixed-nav');
+    document.getElementsByClassName('wrapper')[0].classList.remove('wrapper-nav-offset');
+  }
```

Comment:
When I see a function like this where two blocks are very similar, I try to identify which parts are actually unique. That can help make it easier to understand what's actually important for future readers.

Also, when I notice that two variables/classes always change together, it makes me wonder: Should we just merge the two variables? Or maybe introduce a third variable? In this case, maybe you should have a single "sticky" class on `<body>` which does both things.

Scroll events can fire very rapidly. It's good to think about both the complexity of the work you are doing in here (traversing the entire DOM tree twice to look for nodes) and also if you are doing unnecessary work. Often you can use some kind of stateful flag to track if there is work to do or not.

```
let navIsSticky = false;
window.addEventListener('scroll', function () {
  const navShouldBeSticky = window.scrollY > mobileOffset;
  if(navShouldBeSticky == navIsSticky) {
    // Nothing to to do.
    return;
  }
  document.body.classList.toggle('sticky', navShouldBeSticky);
  navIsSticky = navShouldBeSticky;
});
```

Reply:
Ok yes. I see I'm doing a whole lot of DOM traversing and much of the time adding or removing classes that are already added or removed. :) I will pay with this.